Ark.Tools.FtpClient.ArxOne/FtpClientPoolArxOneWithSocks.cs

```diff
@@ -7,51 +7,57 @@
 using System;
 using System.Net;
 using System.Net.Sockets;
+using System.Text;
 
 namespace Ark.Tools.FtpClient
 {
     public class FtpClientPoolArxOneWithSocks : FtpClientPoolArxOne
     {
         private readonly ISocksConfig _config;
+        private static FtpConfig _ftpConfig = new FtpConfig();
+        private FtpClientParameters _ftpClientParameters = new FtpClientParameters();
 
-        public FtpClientPoolArxOneWithSocks(ISocksConfig config, int maxPoolSize, FtpConfig ftpConfig)
-            : base(maxPoolSize, ftpConfig)
+        public FtpClientPoolArxOneWithSocks(ISocksConfig config, int maxPoolSize, Action<FtpConfig, FtpClientParameters> ftpParameters)
+            : base(maxPoolSize, ftpParameters)
         {
             this._config = config;
+
+            ftpParameters.Invoke(_ftpConfig, _ftpClientParameters);
         }
 
         private protected override ArxOne.Ftp.FtpClient _getClient()
         {
-            var client = new ArxOne.Ftp.FtpClient(this.Uri, this.Credentials, new FtpClientParameters()
+            var ftpClientParameters = this._ftpClientParameters;
```

Comment:
this is a change in lifecycle and is bad as the same instance of ftpParameters is shared amoung different instances of FtpClient and mutated on each creation.